app/controllers/departments_controller.rb

```diff
@@ -0,0 +1,46 @@
+class DepartmentsController < ApplicationController
+
+  def index
+    @departments = Department.all
+  end
+
+  def show
+    @department = Department.find(params[:id])
```

Comment:
user before_action if you do the same work for a few endpoints at the beginning.